src/main/java/com/minecolonies/coremod/client/gui/map/WindowColonyMap.java

```diff
@@ -0,0 +1,337 @@
+     * @return
+     */
+    private void updateBuildingView(final IBuildingView buildingView)
+    {
+        ItemIcon uiBuilding = buildings.get(buildingView);
+        if (uiBuilding == null)
+        {
+            uiBuilding = new ItemIcon();
+
+            uiBuilding.setID(buildingView.getID().toShortString());
+            uiBuilding.setSize(11 + buildingView.getBuildingLevel(), 11 + buildingView.getBuildingLevel());
+            final ItemStack item = buildingView.getBuildingType().getBuildingBlock().asItem().getDefaultInstance();
+            uiBuilding.setItem(item);
+            dragView.addChild(uiBuilding);
+
+            AbstractTextBuilder.TooltipBuilder tooltip = PaneBuilders.tooltipBuilder();
+            tooltip.hoverPane(uiBuilding)
+              .append(window.getScreen().getTooltipFromItem(item).get(0)).append(new StringTextComponent(" : " + buildingView.getBuildingLevel()))
+              .appendNL(new StringTextComponent("Coordinates: " + buildingView.getID().getX() + "X, " + buildingView.getID().getZ() + "Z"))
+              .appendNL(new StringTextComponent("Citizens: " + buildingView.getAllAssignedCitizens().size()));
+
+            for (int id : buildingView.getAllAssignedCitizens())
+            {
+                final ICitizenDataView dataView = building.getColony().getCitizen(id);
+                if (dataView != null)
+                {
+                    tooltip.appendNL(new StringTextComponent(dataView.getName()));
+                }
+            }
+            tooltip.build();
+
+            uiBuilding.setVisible(true);
+
+            dragView.removeChild(this.buildings.get(buildingView));
+            this.buildings.put(buildingView, uiBuilding);
+        }
+
+        uiBuilding.setPosition(worldPosToUIPos(buildingView.getID()).getX(), worldPosToUIPos(buildingView.getID()).getZ());
+    }
+
+    /**
+     * Adds all citizen icons
+     *
+     * @param colony
+     */
+    private void addCitizens(final IColonyView colony)
+    {
+        for (final ICitizenDataView data : colony.getCitizens().values())
+        {
+            final EntityCitizen citizen = (EntityCitizen) colony.getWorld().getEntity(data.getEntityId());
+            if (citizen != null)
+            {
+                final View citizenView = new View();
+                citizenView.setPosition(worldPosToUIPos(citizen.blockPosition()).getX(), worldPosToUIPos(citizen.blockPosition()).getZ());
+
+                final Image citizenImage = new Image();
+                citizenImage.setImage(((ISimpleModelType) IModelTypeRegistry.getInstance().getModelType(citizen.getModelType())).getTextureIcon(citizen));
+                citizenImage.setSize(4, 4);
+                citizenView.addChild(citizenImage);
+
+                dragView.addChild(citizenView);
+                final AbstractTextBuilder.TooltipBuilder builder = PaneBuilders.tooltipBuilder().hoverPane(citizenView).paragraphBreak().append(citizen.getDisplayName());
```

Comment:
Probably don't need first line to be empty
```suggestion
                final AbstractTextBuilder.TooltipBuilder builder = PaneBuilders.tooltipBuilder().hoverPane(citizenView).append(citizen.getDisplayName());
```